griptape/chat_demo/chat.py

```diff
@@ -1,68 +1,112 @@
+    azure_ad_token_provider=lambda: azure_credential.get_token(
+        "https://cognitiveservices.azure.com/.default"
+    ).token,
+    azure_endpoint=os.environ["AZURE_OPENAI_DEFAULT_ENDPOINT"],
+)
+
+
+    
 @define
 class Chat:
-    open_ai_embedding_driver = OpenAiEmbeddingDriver ()
-    vector_store_driver= LocalVectorStoreDriver (embedding_driver=open_ai_embedding_driver)
-    open_ai_prompt_driver= OpenAiChatPromptDriver(model="gpt-4o")
-
-    agent: Agent = field(
-        default=Factory( 
-            lambda self: self.pdf_interact_with_Agent([]),
-            takes_self=True
-        )
+    # Creates a ruleset for the structure 
+    personality_ruleset = Ruleset(
+        name="Personality",
+        rules=[
+            Rule(f"You are a chat bot and your name is 'EDBob'."),
+            Rule(
+                "You speak in a casual but professional manner."
+            ),
+            Rule(
+                "Don't identify as an AI. You are able to have an opinion and have feelings."
+            ),
+            Rule("You are friendly, happy, and generally positive."),
+        ],
     )
-   
-    vector_query_engine= VectorQueryEngine(
-        prompt_driver=open_ai_prompt_driver,
-        vector_store_driver= vector_store_driver,
+    vector_store_client_ruleset = Ruleset(
+        name="Vector Store Client",
+        rules=[
+            Rule("Always search the Vector Store Client."),
+            Rule("Only use the output of the Vector Store Client tool if it is relevant."),
+            Rule("Search the Vector Store Client tool with keywords, not complete sentences or questions."),
+            Rule("Do not add terms to the search like product, details, or price"),
+        ],
     )
 
-    def send_message(self, message: str) -> Any:
-        return self.agent.run(message).output.value
+    knowledge_base_id = os.environ.get("KNOWLEDGE_BASE_ID")
+    gt_cloud_api_key = os.environ.get("GT_CLOUD_API_KEY")
+
+    # Takes cloud base url from the environment variable, if not found, defaults to the cloud url
+    gt_cloud_base_url = os.environ.get("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")
     
-    def upload_pdf(self, file: TemporaryFile) -> str:
-        loaded_file= PdfLoader().load(load_file(file.name))
-        namespace = os.path.basename(file.name)
-        
-        self.vector_query_engine.vector_store_driver.upsert_text_artifacts(
-         {namespace: loaded_file}
-        )
-        vector_store_tool = VectorStoreClient (
-            description=f"Contains information about a PDF with name {namespace}.",
-            query_engine=self.vector_query_engine,
-            namespace=namespace,
-        ) 
-        self.agent = self.pdf_interact_with_Agent(
-            [vector_store_tool]
-        )
-        return file.name
+    # How do I use the vector store driver in Skatepark or locally 
+    vector_store_driver = GriptapeCloudKnowledgeBaseVectorStoreDriver(
```

Comment:
For dev, you may want to use a more simple agent that doesn't have this cloud dependent vector store driver